tests/rptest/tests/franz_go_verifiable_test.py

```diff
@@ -83,3 +86,123 @@ def test_with_all_type_of_loads(self):
+                'cloud_storage_api_endpoint_port': 9000,
+                'cloud_storage_cache_size': self.segment_size * 10
+            })
+
+        self.redpanda_nodes = self.redpanda.nodes
+
+        self.s3client = S3Client(
+            region=self.s3_region,
+            access_key=self.s3_access_key,
+            secret_key=self.s3_secret_key,
+            endpoint=f"http://{self.s3_host_name}:9000",
+            logger=self.logger,
+        )
+        self.s3client.create_bucket(self.s3_bucket_name)
+
+        self._node_for_franz_go = ctx.cluster.alloc(
+            ClusterSpec.simple_linux(1))
+
+        self._producer = FranzGoVerifiableProducer(ctx, self.redpanda,
+                                                   self.topic,
+                                                   SiWithFranzGoTest.MSG_SIZE,
+                                                   1000000000,
+                                                   self._node_for_franz_go)
+        self._seq_consumer = FranzGoVerifiableSeqConsumer(
+            ctx, self.redpanda, self.topic, SiWithFranzGoTest.MSG_SIZE,
+            self._node_for_franz_go)
+        self._rand_consumer = FranzGoVerifiableRandomConsumer(
+            ctx, self.redpanda, self.topic, SiWithFranzGoTest.MSG_SIZE, 10000,
+            20, self._node_for_franz_go)
+
+    def tearDown(self):
+        failed_deletions = self.s3client.empty_bucket(self.s3_bucket_name)
+        assert len(failed_deletions) == 0
+        self.s3client.delete_bucket(self.s3_bucket_name)
+
+    # In the future producer will signal about json creation
+    def _create_json_file(self):
+        small_producer = FranzGoVerifiableProducer(self.test_context,
+                                                   self.redpanda, self.topic,
+                                                   SiWithFranzGoTest.MSG_SIZE,
+                                                   10000,
+                                                   self._node_for_franz_go)
+        small_producer.start()
+        small_producer.wait()
+
+    @cluster(num_nodes=5)
+    def test_with_all_type_of_loads_and_si(self):
+
+        rpk = RpkTool(self.redpanda)
+        rpk.alter_topic_config(self.topic, 'redpanda.remote.write', 'true')
+        rpk.alter_topic_config(self.topic, 'redpanda.remote.read', 'true')
+        rpk.alter_topic_config(self.topic, 'retention.bytes',
+                               str(self.segment_size * 2))
+
+        # Need create json file for consumer at first
+        self._create_json_file()
+
+        # Verify that we really enabled shadow indexing correctly, such
+        # that some objects were written
+        objects = list(self.s3client.list_objects(self.s3_bucket_name))
+        assert len(objects) > 0
+        for o in objects:
+            self.logger.info(f"S3 object: {o.Key}, {o.ContentLength}")
+
+        self._producer.start()
+
+        # Produce some data
+        time.sleep(10)
```

Comment:
prefer to use `wait_until` to wait until data the amount of data you want is available. or, what is often easier, is to place your consumer in a retry loop. adding the sleep can mask races and other problems, as well as making tests unnecessary slow.